Source/WebCore/platform/graphics/PixelBuffer.cpp

```diff
@@ -48,13 +48,16 @@ bool PixelBuffer::supportedPixelFormat(PixelFormat pixelFormat)
     return false;
 }
 
-CheckedUint32 PixelBuffer::computeBufferSize(const PixelBufferFormat& format, const IntSize& size)
+CheckedUint32 PixelBuffer::computeBufferSize(PixelFormat pixelFormat, const IntSize& size)
 {
-    ASSERT_UNUSED(format, supportedPixelFormat(format.pixelFormat));
+    ASSERT_UNUSED(pixelFormat, supportedPixelFormat(pixelFormat));
 
     constexpr unsigned bytesPerPixel = 4;
+    auto bufferSize = CheckedUint32 { size.width() } * size.height() * bytesPerPixel;
+    if (!bufferSize.hasOverflowed() && bufferSize.value() > std::numeric_limits<int32_t>::max())
+        bufferSize.overflowed();
+    return bufferSize;
```

Comment:
Why do we need to ensure the computed size fits within an `int32_t` too? I see that the caller does that check currently, but what relies on that?

FWIW you could also do this with:

```
CheckedUInt32 PixelBuffer::computeBufferSize(...)
{
    ASSERT_UNUSED(...);

    constexpr unsigned bytesPerPixel = 4;
    return CheckedInt32 { CheckedUInt32 { size.width() } * size.height() * bytesPerPixel };
}
```

if you think that's just as clear. (I assume you moved away from `area()` since that also supports computing sizes involving negative dimensions.)

Reply:
> I see that the caller does that check currently, but what relies on that?

 I suppose it's a succint way of saying width > 0 && height > 0. Some ImageData(-5, 5) kind of tests fail without it.

Your proposal without clarifying comment is easy to mistake for a mistake: "Somebody forgot one U here ".

How about writing it explicitly like in the patch?

Reply:
Scrap that, [fast/canvas/canvas-getImageData-invalid-result-buffer-crash.html](https://github.com/WebKit/WebKit/blob/main/LayoutTests/fast/canvas/canvas-getImageData-invalid-result-buffer-crash.html) fails with naive iffing